crates/hc/src/external_subcommands.rs

```diff
@@ -0,0 +1,63 @@
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+};
+
+use crate::builtin_commands;
+
+/// List all runnable commands
+pub fn list_external_subcommands() -> Vec<String> {
+    let prefix = "hc-";
+    let suffix = env::consts::EXE_SUFFIX;
+    let mut commands = Vec::new();
+
+    let builtin_cmds = builtin_commands();
+
+    for dir in search_directories() {
```

Comment:
Does this appreciably increase the time it takes to run `hc`? Because this will get executed every time, right? Not just when running `hc` without subcommands? Or does the laziness of this get respected by StructOpt?

Reply:
Hum that's a good question actually, I don't know... The experience in my laptop right now is instantianious. How would we go about testing that?

Reply:
oh I know how to test it. Hang on

Reply:
Yeah it *will* get executed every time *any* hc command is executed...

Reply:
Hmm, well it *should* be OK right? Especially if this is how `cargo` does it.

Reply:
Mmm I would think so yeah. If it causes a problem in some systems we can refactor and improve this.